src/panels/scene-viewer/SceneViewer.tsx

```diff
@@ -159,11 +173,32 @@ export const SceneViewer = (props: SceneViewerPropsFromParent) => {
       dataBindingTemplate[undecoratedKey] = selectedPropertyValue;
     }
 
-    const selectedDataBinding = {
-      [DataBindingLabelKeys.entityId]: selectedEntityValue ?? '',
-      [DataBindingLabelKeys.componentName]: selectedComponentValue ?? '',
-      [DataBindingLabelKeys.propertyName]: selectedPropertyValue ?? '',
-    };
+    let selectedDataBinding: Record<string, string> | undefined =
+      selectedEntityValue && selectedComponentValue
+        ? {
+            [DataBindingLabelKeys.entityId]: selectedEntityValue,
+            [DataBindingLabelKeys.componentName]: selectedComponentValue,
+          }
+        : undefined;
+
+    const selectedNode = getSceneNodeByRef(getSelectedSceneNodeRef() || '');
```

Comment:
Do we always use nested function calls to get the selectedNode? What use case is there to use just one of these functions?
Seems like we don't need to return both from `useSceneComposerApi`

Reply:
`getSceneNodeByRef` should be a useful function to be exposed on it's own, so I just make them separate ones